tasks/run_cd4pe_job.rb

```diff
@@ -102,7 +102,7 @@ def initialize(web_ui_endpoint:, job_token:, ca_cert_file: nil, logger:)
 
   def make_request(type, api_url, payload = '')
     connection = Net::HTTP.new(@http_config[:server], @http_config[:port])
-    if @http_config[:scheme] == 'https'
+    if @http_config[:scheme] == 'https' and !@ca_cert_file.nil?
```

Comment:
We need to do the check inside the if statement. If the schema is https, we need to tell ruby's net/http lib to use ssl. I think the if check just needs to go around lines 110.
```
store = OpenSSL::X509::Store.new
store.set_default_paths
if !@ca_cert_file.nil?
  store.add_file(@ca_cert_file)
end
connection.cert_store = store

```

This way we continue to use the system SSL store and verify connections.

I still think there will be edge cases with this approach though - nothing concrete i can think of off the top of my head

Reply:
Yeah, I think I missed something. I think it's that we don't want to create the custom store if we don't have a cert to put in it, so like:
```
    if @http_config[:scheme] == 'https'
      connection.use_ssl = true
      connection.verify_mode = OpenSSL::SSL::VERIFY_PEER
      if @ca_cert_file
        store = OpenSSL::X509::Store.new
        store.set_default_paths
        store.add_file(@ca_cert_file)
        connection.cert_store = store
      end
    end
```